tools/accuracy_checker/custom_evaluators/colorization_v2_evaluator.py

```diff
@@ -0,0 +1,215 @@
+from accuracy_checker.adapters import create_adapter
+from accuracy_checker.data_readers import BaseReader
+from accuracy_checker.config import ConfigError
+from accuracy_checker.preprocessor import PreprocessingExecutor
+from accuracy_checker.metrics import MetricsExecutor
+from accuracy_checker.launcher import create_launcher
+from accuracy_checker.utils import extract_image_representations
+import cv2
+
+class ColorizationEvaluator(BaseEvaluator):
+    def __init__(self, dataset, reader, preprocessing, metric_executor, launcher, test_model, check_model):
+        self.dataset = dataset
+        self.preprocessing_executor = preprocessing
+        self.metric_executor = metric_executor
+        self.launcher = launcher
+        self.test_model = test_model
+        self.check_model = check_model
+        self.reader = reader
+        self._metrics_results = []
+
+    @classmethod
+    def from_configs(cls, config):
+        dataset_config = config['datasets'][0]
+        dataset = Dataset(dataset_config)
+        data_reader_config = dataset_config.get('reader', 'opencv_imread')
+        data_source = dataset_config['data_source']
+        if isinstance(data_reader_config, str):
+            reader = BaseReader.provide(data_reader_config, data_source)
+        elif isinstance(data_reader_config, dict):
+            reader = BaseReader.provide(data_reader_config['type'], data_source, data_reader_config)
+        else:
+            raise ConfigError('reader should be dict or string')
+        preprocessing = PreprocessingExecutor(dataset_config.get('preprocessing', []), dataset.name)
+        metrics_executor = MetricsExecutor(dataset_config['metrics'], dataset)
+        launcher = create_launcher(config['launchers'][0], delayed_model_loading=True)
+        test_model = ColorizationTestModel(config.get('network_info', {}), launcher)
+        check_model = ColorizationCheckModel(config.get('network_info', {}),
+                                           dataset_config.get('preprocessing', {}), launcher)
+        return cls(dataset, reader, preprocessing, metrics_executor, launcher, test_model, check_model)
+
+    def process_dataset(self, stored_predictions, progress_reporter, *args, ** kwargs):
+        self._annotations, self._predictions = ([], []) if self.metric_executor.need_store_predictions else None, None
+        if progress_reporter:
+            progress_reporter.reset(self.dataset.size)
+
+        for batch_id, (dataset_indices, batch_annotation) in enumerate(self.dataset):
+            batch_identifiers = [annotation.identifier for annotation in batch_annotation]
+            batch_input = [self.reader(identifier=identifier) for identifier in batch_identifiers]
+            batch_input = self.preprocessing_executor.process(batch_input, batch_annotation)
+            batch_input, _ = extract_image_representations(batch_input)
+            batch_out = np.array(self.test_model.predict([], batch_input))
```

Comment:
for consistency
```suggestion
            batch_out = np.array(self.test_model.predict(batch_identifiers, batch_input))
``
it will be easier to debug if we will know the image name.`

Reply:
@eaidova done)